project/cmake/modules/FindFFMPEG.cmake

```diff
@@ -1,17 +1,164 @@
-include(ExternalProject)
-file(STRINGS ${CORE_SOURCE_DIR}/tools/depends/target/ffmpeg/FFMPEG-VERSION VER)
-string(REGEX MATCH "VERSION=[^ ]*$.*" FFMPEG_VER "${VER}")
-list(GET FFMPEG_VER 0 FFMPEG_VER)
-string(SUBSTRING "${FFMPEG_VER}" 8 -1 FFMPEG_VER)
-string(REGEX MATCH "BASE_URL=([^ ]*)" FFMPEG_BASE_URL "${VER}")
-list(GET FFMPEG_BASE_URL 0 FFMPEG_BASE_URL)
-string(SUBSTRING "${FFMPEG_BASE_URL}" 9 -1 FFMPEG_BASE_URL)
-
+if(FFMPEG_PATH)
+  set(ENABLE_INTERNAL_FFMPEG OFF)
+endif()
 
-if(ENABLE_INTERNAL_FFMPEG)
+# external FFMPEG
+if(NOT ENABLE_INTERNAL_FFMPEG)
   if(FFMPEG_PATH)
-    message(WARNING "Internal FFmpeg enabled, but FFMPEG_PATH given, ignoring")
+    set(ENV{PKG_CONFIG_PATH} "${FFMPEG_PATH}/lib/pkgconfig")
+    list(APPEND CMAKE_PREFIX_PATH ${FFMPEG_PATH})
+  endif()
+
+  # FFMPEG v3.1
+  set(REQUIRED_FFMPEG_VERSION 3.1)
+  set(FFMPEG_PKGS libavcodec>=57.48.101
```

Comment:
imo those checks are a bit overenginered. note that developers need to be able to build with any version of ffmpeg. if external ffmpeg is used we can expect from that persons that they know what they are doing. hence no version check required.